USFMConverter/Core/Render/RenderDocument.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using USFMConverter.Core.ConstantValue;
+using USFMConverter.Core.Data;
+using USFMToolsSharp;
+using USFMToolsSharp.Models.Markers;
+
+namespace USFMConverter.Core.Render
+{
+    public abstract class RenderDocument
+    {
+        public RenderDocument()
+        {
+            
+        }
+
+        /// <summary>
+        /// Parses the given text files into one USFM Document.
+        /// </summary>
+        /// <param name="files">Text files with USFM format.</param>
+        /// <returns>A USFM Document</returns>
+        public USFMDocument LoadUSFMs(IEnumerable<FileInfo> files)
+        {
+            var usfmDoc = new USFMDocument();
+            var parser = new USFMParser(new List<string> { "s5" });
+
+            foreach(var file in files)
+            {
+                var text = File.ReadAllText(file.FullName);
+                usfmDoc.Insert(parser.ParseFromString(text));
+            }
+
+            return usfmDoc;
+        }
+
+        /// <summary>
+        /// Parses the given text files into one USFM Document asynchronously.
+        /// </summary>
+        /// <param name="files">Text files with USFM format.</param>
+        /// <param name="progressCallback">Call back for progress bar update.</param>
+        /// <returns>A USFM Document</returns>
+        public async Task<USFMDocument> LoadUSFMsAsync(
```

Comment:
This function is not related to the RenderableDocument class.  It should be move into a utilities or file interface class.